### DISCUSSION TITLE
EMMC throughput question

### DISCUSSION THREAD

**Discussion by josh25219** [polarfire-soc/polarfire-soc-documentation#371] [Bare metal embedded software] [answered]
With the EMMC in HS200 mode, theoretical Bandwith max would be 200mhz * 8bit is 200Mbyte/sec.    Using the  MSS_MMC_cq functions to write and read 400Mbytes of data we are seeing much less than 200Mbyte/sec transfer rates.   Is there a limiting factor elsewhere in the MSS?

**Comment by padmaraob** [comment]
No Limiting factor,  Yes data transfer rates are less than 200MB/sec depend on an eMMC device. We have done with Sandisk device for HS200 mode - 185MB/sec for read  and 42MB/sec for write operation.

**Reply by josh25219** [reply]
Thats great info!   Is it the EMMC write operation itself that dictates such a slower write operation compared to the read?

**Reply by padmaraob** [reply]
Yes

**Reply by josh25219** [reply]
Hi,
Which of the 3 example drivers in the bare metal section for mmc were used to do the throughput test that you mention above?   
Also, How did you measure the rate for write and read?

**Reply by padmaraob** [reply, marked as answer]
This was already answered through mail.

Example project "mpfs-emmc-sd-write-read" used for throughput.

Throughput Calculation
****************************
Time count to execute the write function = Timer Load Value – Timer Current value
Time Duration = Time Count * (1/clk_freq)
Throughput = data size * (1/Time Duration)
